org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/classfmt/ClassFileReader.java

```diff
@@ -204,7 +204,7 @@ public static ClassFileReader read(String fileName, boolean fullyInitialize) thr
  * @param fileName	Actual name of the file that contains the bytes, can be null
  *
  * @exception ClassFormatException
- * @Deprecated Use {@link #ClassFileReader(URI, byte[], char[])} where an annotation processor might be in the picture
+ * @deprecated Use {@link #ClassFileReader(URI, byte[], char[])} where an annotation processor might be in the picture
```

Comment:
Coming back here after seeing all the new `@SuppressWarnings("deprecation")` I wonder if deprecating this constructor (in response to my review in #668) was a good idea, given that it is only a conditional deprecation. 
**If** we use deprecation & suppresswarnings, then I recommend to add a comment to each suppress, indicating why the conditional deprecation doesn't hold in that particular case. 
We my need to consult with @jarthana  here.

Reply:
i don't mind in which way we go as long as it is consistent. does not need to be fixed with this pr

Reply:
> does not need to be fixed with this pr

Fine for me, so to avoid unnecessary ping-pong changes please
* file a follow up issue to clarify the underlying question about usage of this constructor, cc'ing Jay and me
* revert changes relating to this half-baked design from the current PR (perhaps replace the broken javadoc element with a FIXME or similar for now).